### DISCUSSION TITLE
My Setup

### DISCUSSION THREAD

**Discussion by kenlpeters** [philosowaffle/peloton-to-garmin#99] [Ideas]
So I wanted to post how I incorporated this into my world and fitness.  

1)  Created a Azure Debian Linux server (free tier)
2)  Installed the peloton-to-garmin software and configured it
3)  Installed Node.Js
4)  Created a NodeJs server and have it running under PM2 on a designated port, response returned Output from calling peloton-to-garmin
5)  I created an Apple Shortcut on my iPhone that calls my URL for the NodeJS server which then executes the peloton-to-garmin py
6)  The call returns the output and sends me a text with the data (just so I can ensure a successful import)
7)  I added the Apple Shortcut to my homepage as an icon.
-----------------------------------------------------------------
After each workout I simply press the icon my phone homepage
Its executes the sync and sends me a text message
I go into Garmin and my workouts are there.
Easy as Pi
-----------------------------------------------------------------
I am sure you could also do this with AWS linux server or an google one.  The trick is to find a free tier or if you are Microsoft Developer (MSDN license) you get $150/month credit.  Use it.

**Comment by philosowaffle** [comment]
That is really cool!  One of the long term ideas I have for P2G version2 is to expose a simple API and a simple web interface that would allow you to easily kick off adhoc sync jobs for a run.  Once the basics of an api and web ui are in place then there are actually a lot of other really cool things that can be done.......eventually. First step is just getting version 2 fully ready for prime time.  Its about 95% there, but still needs some polishing.

**Reply by kenlpeters** [reply]
Excellent - Let me know if I can help in anyway.
